src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -433,6 +434,68 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                 {
                     throw new InvalidOperationException($"Docker exec fail with exit code {execEchoExitCode}");
                 }
+
+                bool setupDockerGroup = executionContext.Variables.GetBoolean("VSTS_SETUP_DOCKERGROUP") ?? StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable("VSTS_SETUP_DOCKERGROUP"), true);
+                if (setupDockerGroup)
+                {
+                    executionContext.Output(StringUtil.Loc("AllowContainerUserRunDocker", containerUserName));
+                    // Get docker.sock group id on Host
+                    string dockerSockGroupId = (await ExecuteCommandAsync(executionContext, "stat", $"-c %g /var/run/docker.sock")).FirstOrDefault();
+
+                    // We need to find out whether there is a group with same GID inside the container
+                    string existingGroupName = null;
+                    List<string> groupsOutput = new List<string>();
+                    int execGroupGrepExitCode = await _dockerManger.DockerExec(executionContext, container.ContainerId, string.Empty, $"bash -c \"cat /etc/group\"", groupsOutput);
+                    if (execGroupGrepExitCode != 0)
+                    {
+                        throw new InvalidOperationException($"Docker exec fail with exit code {execGroupGrepExitCode}");
+                    }
+
+                    if (groupsOutput.Count > 0)
+                    {
+                        // check all potential username that might match the UID.
```

Comment:
Should this be "check all potential *groups* that might match the *GID*? Otherwise LGTM